### ISSUE TITLE
Appium 1.5.0-beta11 start: TypeError: Path must be a string. Received undefined

### ISSUE THREAD

**Issue by rolandoldengarm** [appium/appium#6065]
I've installed appium@beta which may fix another issue with Chromedriver. However, after installing it and starting it, I get the following error:

````
$ appium
TypeError: Path must be a string. Received undefined
    at assertPath (path.js:8:11)
    at Object.win32.resolve (path.js:130:5)
    at Object.<anonymous> (lib/ios-crash-log.js:8:24)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (index.js:1:35)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (lib/driver.js:11:51)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
````
I've tried reinstalling it, but it does not make a difference.

**Comment by rolandoldengarm** [comment]
I've done more investigation. Update all my global NPM packages. Issue happens with both the npm package appium@beta as well as when building from source locally.

**Comment by imurchie** [comment]
I'm unable to get this to happen, either through npm or source. :(

My only thought is that something is not updated from an older install. Can you delete the whole `node_modules/appium` directory, and reinstall? Also, can you run `npm cache clean`?

**Comment by rolandoldengarm** [comment]
I've done that a couple of tried. Also redownloaded from GIT, but does not work. Also now trying node-inspector to debug, but I then only see a white screen or node-debug crashes...
Very keen to get this fixed as this is blocking our tests going forward. Anything else I can check? The stack trace doesn't make sense to me, is there any way to see which path is null? Can it be an environment variable I don't have?

**Comment by imurchie** [comment]
I don't think there should be any environment variables that are set.

What version of node are you using? And npm? OS?

**Comment by rolandoldengarm** [comment]
node 4.1.0, npm 2.7.4, Windows 10

Not sure if it is related, but when installing node packages, e.g. when installing appium 1.4 I get this error: 
````
 if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (rebuild)
 configure error
gyp ERR! stack Error: spawn c:\python27 ENOENT
gyp ERR! stack     at exports._errnoException (util.js:837:11)
````
But everything else just works fine, so not sure if it is related.

**Comment by rolandoldengarm** [comment]
Using nvm, I've switched to both 4.2.6 and 5.5.0. No difference :(

**Comment by imurchie** [comment]
I don't have access to a Windows machine right at this moment. Since you
have it "running" from source, could you try adding a console log to the
beginning of 'lib/main.js' to see if it is getting in there at all? You
will need to run 'gulp transpire' after modifying anything.

Also, from source could you try running Appium by using 'node
build/lib/main.js' ?

Thanks!

**Comment by rolandoldengarm** [comment]
I've added "console.log("MAIN.JS") as the first line in main.js, just below  #!/usr/bin/env node
Then, ran gulp transpile (not transpire, that was a typo I guess).

Finally ran node buid/lib/main.js, same error:
````
$ node build/lib/main.js                          
TypeError: Path must be a string. Received undefin
    at assertPath (path.js:8:11)                  
    at Object.win32.resolve (path.js:130:5)       
    at Object.<anonymous> (lib/ios-crash-log.js:8:
    at Module._compile (module.js:410:26)         
    at Object.Module._extensions..js (module.js:41
    at Module.load (module.js:344:32)             
    at Function.Module._load (module.js:301:12)   
    at Module.require (module.js:354:17)          
    at require (internal/module.js:12:17)         
    at Object.<anonymous> (index.js:1:35)         
    at Module._compile (module.js:410:26)         
    at Object.Module._extensions..js (module.js:41
    at Module.load (module.js:344:32)             
    at Function.Module._load (module.js:301:12)   
    at Module.require (module.js:354:17)          
    at require (internal/module.js:12:17)         
    at Object.<anonymous> (lib/driver.js:11:51)   
    at Module._compile (module.js:410:26)         
    at Object.Module._extensions..js (module.js:41
    at Module.load (module.js:344:32)             
    at Function.Module._load (module.js:301:12)   
    at Module.require (module.js:354:17)          
    at require (internal/module.js:12:17)         
    at Object.<anonymous> (lib/server.js:3:69)    
    at Module._compile (module.js:410:26)         
    at Object.Module._extensions..js (module.js:41
    at Module.load (module.js:344:32)             
    at Function.Module._load (module.js:301:12)   
````
Looks like it's loading one of the modules which fails? 

Thanks for the quick replies. Hopefully we can solve this.

**Comment by imurchie** [comment]
Yes. I see where it seems to be going wrong. I can fix it, but am on a bus
right now, on my phone. I'll be on my computer on a couple of hours and
will fix the issue.

In the meantime, you won't be using iOS on Windows, so you can go into
'lib/appium.js' and comment out the references to 'IosDriver'. Then 'gulp
transpile' (sorry for the autocorrect).  Crossing fingers that will work.

Sorry for the bother.

**Comment by imurchie** [comment]
It turns out that if you set an environment variable 'HOME' to something it
ought to work. It's a bug, and I will fix it, but in the meantime that
ought to work.

**Comment by rolandoldengarm** [comment]
AWESOME! Setting "HOME" fixed it :)

**Comment by imurchie** [comment]
Ugh. It's a bad error. I'll go through the rest of the codebase to see if
there are other instances of such things.

**Comment by imurchie** [comment]
But happy that there is a temporary fix. :)

**Comment by abacaj** [comment]
Same issue - how does the fix work? Where do you set "HOME"... pretty annoying can't even get the android to start.

**Comment by imurchie** [comment]
This should be fixed in the latest release. What is the output when you try to run it?

**Comment by abacaj** [comment]
Same exact output as the previous user, but the issue was related to aapt, android asset packaging tool - it was looking for the wrong version. I figured it out because running `aapt` in my terminal gave me a `it appears you do not have blah blah installed`.

**Comment by imurchie** [comment]
Can you post the output anyway, so we can try to get a better error message should this happen again?

**Comment by abacaj** [comment]
Now I am getting a different error:

`error: logcat capture failed: spawn "/usr/local/bin/adb" ENOENT`

**Comment by imurchie** [comment]
Does `/usr/local/bin/adb` exist? Where is your Android SDK installed?

**Comment by abacaj** [comment]
yes running `/usr/local/bin/adb` brings up the android debug bridge command

**Comment by imurchie** [comment]
Can you post, in a [gist](https://gist.github.com/), the whole server output so we can see what is going on?

**Comment by abacaj** [comment]
cool - fixed it with `export ANDROID_HOME=/usr/local/Cellar/android-sdk/24.4.1_1`

**Comment by lujian2036** [comment]
whether this bug is fix in 1.6.0 version?i still suffer this problem
node -v : v6.9.1
npm -v : 3.10.8
window7 *64
now i set HOME variable to solve this problem  temporary 
i install appium by  npm install -g appium ,while set registry=https://registry.npm.taobao.org,because gfw block download chromedriver

**Comment by imurchie** [comment]
This is fixed.
